components/merkledb/src/proof_map_index/mod.rs

```diff
@@ -202,11 +209,20 @@ impl BinaryAttribute for Option<ProofPath> {
     }
 }
 
-impl<T, K, V> ProofMapIndex<T, K, V>
+/// Hashed variant of the `ProofMapIndex`, useful for arbitrary keys that implement
+/// `ObjectHash` trait.
+pub type HashedProofMap<T, K, V> = ProofMapIndex<T, K, V, Hashed>;
```

Comment:
I don't like inconsistency of all the indices having "*Index" postfix and "ProofMap" going out of this convention.